### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by izdeby**
Updated README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,95 @@
-#  Under construction
+## Motivation
 
-Currently migrating code from other sources.
+Many fields manipulate collections of Tensors of different shapes. For example, paragraphs of text, images of different sizes or audio files of different lengths. We don't have a first class abstraction that enables the concurrent manipulation of collections of this type of data. We further often need to batch arbitrary data and operations for efficiency.
 
-![alt text](assets/sign.png "Construction sign")
+## Decription
+
+NestedTensor is an abstraction of torch tensor which allows working with data of different sizes and length.
```
Comment (cpuhrsch) [inline README.md]: *generalization instead of abstraction

File: README.md [hunk 2]
```diff
@@ -1,5 +1,95 @@
...
 
-Currently migrating code from other sources.
+Many fields manipulate collections of Tensors of different shapes. For example, paragraphs of text, images of different sizes or audio files of different lengths. We don't have a first class abstraction that enables the concurrent manipulation of collections of this type of data. We further often need to batch arbitrary data and operations for efficiency.
 
-![alt text](assets/sign.png "Construction sign")
+## Decription
+
+NestedTensor is an abstraction of torch tensor which allows working with data of different sizes and length. 
+
+In general, there are two cases for which NestedTensors provide computational representations.
```
Comment (cpuhrsch) [inline README.md]: I'd split out the constraints into a subsection

File: README.md [hunk 3]
```diff
@@ -1,5 +1,95 @@
...
+**Lists of Tensors**
+Each Tensor constituent of the list it represents, if any, must be of its dtype, layout and device. The dimension of a constituent Tensor must be one less than the dimension of the NestedTensor. An empty list of Tensors yields a NestedTensor of dimension one.
+
+**Lists of NestedTensors**
+Each constituent NestedTensor must be of its dtype, layout and device. The dimension of a constituent NestedTensor must be one less than the dimension of the NestedTensor.
+
+## Prerequisites
+
+You will need this packages to be installed:
+
```
Comment (cpuhrsch) [inline README.md]: Qualify what each package is needed for. The raw repo only requires pytorch, the example notebooks require torchvision etc.

File: README.md [hunk 4]
```diff
@@ -1,5 +1,95 @@
...
+```
+a = torch.tensor([1])
+b = torch.tensor([[2, 2],
+                  [3, 3],
+                  [4, 4],
+                  [5, 5]])
+nt2 = torch.nested_tensor([[a],[b]])
+```
+
+The level of nesting is inferred from the input. The constructor always copies. Whatever you pass into the constructor will share no data with what the constructor returns. This matches torch.tensor's behavior.
```
Comment (cpuhrsch) [inline README.md:70]: There's also ```as_nested_tensor``` that allows a non-copying wrapper. Not sure that's too much here.

File: README.md [hunk 5]
```diff
@@ -1,5 +1,95 @@
...
+For all other ops, please, see examples.
+
+
+## The tensorwise decorator
+The nestedtensor package allows the user to decorate existing functions with a tensorwise decorator. This decorator lifts the given function to check for NestedTensor arguments and apply recursively apply it to their constiuents with all other arguments untouched.
+
+Decorating the function as tensorwise does not affect its behavior with respect to non-NestedTensor arguments. In particular, the tensorwise decorator will search all arguments for a NestedTensor and if none is found dispatch to exactly the given function.
+
+```
+@tensorwise()
```
Comment (cpuhrsch) [inline README.md:98]: I'd expand this a bit more to qualify as a full example

File: README.md [hunk 6]
```diff
@@ -1,5 +1,95 @@
...
+       [torch.rand(6, 7)] \
+      ]
+
+b = torch.nested_tensor(a)
+b1 = b.unbind() # Tuple of 2 NestedTensors
+b2 = b1[0].unbind() # Tuple of 2 Tensors
+```
+
+### Other Ops
+For all other ops, please, see examples.
```
Comment (cpuhrsch) [inline README.md:89]: There's a codegen function that gives you a list of all bindings, but I'm waiting on ```__torch_function__``` to give us an easy way of showing support. Really we shouldn't use monkeypatching. Here is the branch to use torch function: https://github.com/cpuhrsch/nestedtensor/tree/torchfunction

Comment (izdeby) [reply README.md:89]: I think that if its a branch, we should not put it into a doc. its like telling people that pytroch has nested tensors if you pick the right branch.

Comment (cpuhrsch) [reply README.md:89]: Yes, this was just for your information.

File: README.md [hunk 7]
```diff
@@ -1,5 +1,110 @@
...
 
-Currently migrating code from other sources.
+## Motivation
 
-![alt text](assets/sign.png "Construction sign")
+Many fields manipulate collections of Tensors of different shapes. For example, paragraphs of text, images of different sizes or audio files of different lengths. We don't have a first class abstrageneralizationction that enables the concurrent manipulation of collections of this type of data. We further often need to batch arbitrary data and operations for efficiency.
+
+## Decription
+
+NestedTensor is an generalization of torch tensor which allows working with data of different sizes and length.
```
Comment (cpuhrsch) [inline README.md:9]: *a generalization

**Pull request closed without merging**